lib/controllers/Controller.js

```diff
@@ -54,6 +58,28 @@ Controller.prototype.handleRequest = function (request, response, next, settings
   }
 };
 
+// Get host and protocol from HTTP's Forwarded header
+Controller.prototype._getForwarded = function (request) {
+  if (!request.headers.forwarded)
+    return {};
+  try {
+    var forwarded = parseForwarded(request.headers.forwarded)[0];
+    return {
+      protocol: forwarded.proto + ':',
+      host: forwarded.host,
+    };
+  }
+  catch (error) { return {}; }
```

Comment:
What cases are caught here? The fact that `parseForwarded` throws an error, returns `undefined`, or returns an empty array? If not the first case, then we should rewrite this without `try / catch`.

Reply:
It caught a possible "Error" (literally) exception. I thought that since that error message wasn't really useful maybe it would be best to write a "cannot parse forwarded header: ...." in the warnings but then maybe it wasn't really useful anyway. I can add a "console.warn" if you want?

Reply:
Thanks, but that's fine. I was just wondering what we were catching here; an actual `Error` from `parseForwarded` is good.

Reply:
Or maybe we can return a BadRequest?

Reply:
`{}` is good for me, this gives the behavior we want upstream.